login1/dbus.go

```diff
@@ -74,6 +74,123 @@ func (c *Conn) initConnection() error {
 	return nil
 }
 
+// Session object definition.
+type Session struct {
+	ID   string
+	UID  uint32
+	User string
+	Seat string
+	Path dbus.ObjectPath
+}
+
+// User object definition.
+type User struct {
+	UID  uint32
+	Name string
+	Path dbus.ObjectPath
+}
+
+func (s Session) toInterface() []interface{} {
+	return []interface{}{s.ID, s.UID, s.User, s.Seat, s.Path}
+}
+
+func sessionFromInterfaces(session []interface{}) (*Session, error) {
+	id := session[0].(string)
+	uid := session[1].(uint32)
+	user := session[2].(string)
+	seat := session[3].(string)
+	path, ok := session[4].(dbus.ObjectPath)
```

Comment:
Sorry, I wasn't clear enough in my previous review. Please make sure that all type assertions (i.e. also fields 0 to 3) are checked, like you did for the last field.